lib/iris/fileformats/pp_rules.py

```diff
@@ -597,9 +597,12 @@ def date2hours(t):
         coords_and_dims.append((DimCoord(t1_epoch_hours, standard_name='time', units=epoch_hours_unit), None))
         coords_and_dims.append((DimCoord(t2_epoch_hours, standard_name='forecast_reference_time', units=epoch_hours_unit), None))
 
+    # Don't add time coordinates when years are zero, months match, and
+    # lbtim.ib == 2. These are handled elsewhere.
     if \
             (lbtim.ib == 2) and \
             (lbtim.ic in [1, 2, 4]) and \
+            (not (t1.year == 0 and t2.year == 0 and t1.month == t2.month)) and \
```

Comment:
what do we do in the cases:
 one year is set and the other is zero?
 both years are zero, but the months do not match

at present we would run through this code path.

I think we are better to protect this path more carefully, perhaps
```
(not (t1.year == 0 or t2.year == 0))
```
as year 0 is undefined in the gregorian calendar

this would enable the code path at line 910 to run when required, this path to run when safe, and if in doubt, don't build bonkers datetimes